### PR TITLE
Extend DRY support for persistence and Ingress

### PR DESCRIPTION
**Pull request by Asif-git03**
### What does this PR do?

 Add tpl support for persistence.storageClassName in home-pvc.yaml
--> Add tpl support for controller.ingress parameters(ingressClassName, annotations, hostname) in jenkins-controller-ingress.yaml

- Fixes #

If you modified files in the `./charts/jenkins/` directory, please also include the following:

```[tasklist]
### Submitter checklist
- [x] I bumped the "version" key in `./charts/jenkins/Chart.yaml`.
- [x] I added a new changelog entry to `./charts/jenkins/CHANGELOG.md`.
- [ ] I followed the [technical requirements](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#technical-requirements).
- [ ] I ran `.github/helm-docs.sh` from the project root.
```

### CODE REVIEW

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
Hi please bump the chart version and add a changelog entry, LGTM apart from that

File: charts/jenkins/templates/home-pvc.yaml [hunk 1]
```diff
@@ -34,7 +34,7 @@ spec:
 {{- if (eq "-" .Values.persistence.storageClass) }}
   storageClassName: ""
 {{- else }}
-  storageClassName: "{{ .Values.persistence.storageClass }}"
+  storageClassName: "{{ tpl .Values.persistence.storageClass }}"
```
Comment (CeddaerrixGE) [inline charts/jenkins/templates/home-pvc.yaml:37]: Syntax from https://helm.sh/docs/howto/charts_tips_and_tricks/#using-the-tpl-function not respected. the `tpl` function take 2 parameters. It should be:
`storageClassName: "{{ tpl .Values.persistence.storageClass . }}"`

File: charts/jenkins/templates/jenkins-controller-ingress.yaml [hunk 2]
```diff
@@ -23,12 +23,12 @@ metadata:
...
 {{- if .Values.controller.ingress.annotations }}
   annotations:
-{{ toYaml .Values.controller.ingress.annotations | indent 4 }}
+{{ tpl (toYaml .Values.controller.ingress.annotations) . | indent 4 }}
 {{- end }}
   name: {{ template "jenkins.fullname" . }}
 spec:
 {{- if .Values.controller.ingress.ingressClassName }}
-  ingressClassName: {{ .Values.controller.ingress.ingressClassName | quote }}
+  ingressClassName: {{ tpl .Values.controller.ingress.ingressClassName | quote }}
```
Comment (CeddaerrixGE) [inline charts/jenkins/templates/jenkins-controller-ingress.yaml:31]: The `tpl` function syntax is not respected: 2 parameters required

File: charts/jenkins/CHANGELOG.md [hunk 3]
```diff
@@ -14,6 +14,10 @@ Those entries include a reference to the git commit to be able to get more detai
 
 ## 5.7.15
```
Comment (CeddaerrixGE) [inline charts/jenkins/CHANGELOG.md]: Wrong version

**Review by CeddaerrixGE — CHANGES_REQUESTED** [review]

**Comment by CeddaerrixGE** [comment]
BTW - You are missing unit tests for this implementation

**Review by timja — APPROVED** [review]

**Pull request merged**